File: src/test/java/kitchenpos/products/application/tobe/ProductTest.java
package kitchenpos.products.application.tobe;


import kitchenpos.products.tobe.domain.ProductService;
import kitchenpos.products.tobe.domain.entity.Product;
import kitchenpos.products.tobe.domain.infra.InMemoryProductRepository;
import kitchenpos.products.tobe.domain.vo.DisplayedName;
import kitchenpos.products.tobe.domain.vo.Price;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;

public class ProductTest {

    @DisplayName("상품을 등록할 수 있다")
    @Test
    void registerNewTest() {
        Price price = new Price(10);
        DisplayedName displayedName = new DisplayedName("name");

        ProductService service = new ProductService(new InMemoryProductRepository());
        service.register(displayedName, price);

        List<Product> pdList = service.getList();
        assertThat(pdList).hasSize(1);
    }

    @DisplayName("상품에는 가격과 이름이 필요로 하다")
    @Test
    void EntityPropertyTest() {
        Price price = new Price(100);
        DisplayedName displayedName = new DisplayedName("my-product");
        Product pd = new Product(displayedName, price);
        Assertions.assertNotNull(pd);
    }
}

File: src/test/java/kitchenpos/products/application/tobe/ProductTest.kt
package kitchenpos.products.application.tobe

import kitchenpos.common.price
import kitchenpos.products.application.FakeProductNameValidatorService
import kitchenpos.products.tobe.domain.Product
import kitchenpos.products.tobe.domain.productPrice
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.math.BigDecimal

class ProductTest {
    @Test
    fun `정상적인 상품 생성`() {
        val product = Product(
            displayedName = "양념감자",
            price = BigDecimal.valueOf(4000).price().productPrice(),
            FakeProductNameValidatorService()
        )

        val `4천원` = BigDecimal.valueOf(4000).price().productPrice()

        assertThat(product.displayedName).isEqualTo("양념감자")
        assertThat(product.price).isEqualTo(`4천원`)
    }
}
